src/ga/population.rs

```diff
@@ -139,6 +141,46 @@ impl PopulationGenerator<Vec<bool>> for BitStrings {
   }
 }
 
+/// Implements [PopulationGenerator] trait. Can be used with genetic algorithm.
+///
+/// Generates random permutations of given vector.
+/// Permutations can be repeated
+pub struct RandomPermutations<GeneT: Copy> {
+  genes: Vec<GeneT>,
+}
+
+impl<GeneT: Copy> RandomPermutations<GeneT> {
+  /// Returns [RandomPermutations] population generator.
+  ///
+  /// ### Arguments
+  ///
+  /// * `genes` - Vector which will be permuted
+  pub fn new(genes: Vec<GeneT>) -> Self {
+    RandomPermutations { genes }
+  }
+}
+
+impl<GeneT: Copy + Debug + Sync + Send> PopulationGenerator<Vec<GeneT>> for RandomPermutations<GeneT> {
+  /// Generates vector of `count` random permutations from stored genes.
+  /// Repeated individual are possible.
+  ///
+  /// ### Arguments
+  ///
+  /// * `count` -- Number of random permutations to generate
+  fn generate(&self, count: usize) -> Vec<Individual<Vec<GeneT>>> {
+    let mut population: Vec<Individual<Vec<GeneT>>> = Vec::with_capacity(count);
+    let rng = &mut rand::thread_rng();
+
+    for _ in 0..count {
+      let mut genome = self.genes.clone();
+      genome.shuffle(rng);
+      population.push(Individual::from(genome))
+    }
+
+    population
+  }
+}
```

Comment:
There are surely much more efficient ways of generating permutations than this. 
I guess, we'll go with this implementation for now, but before this PR is merged please create a separate issue with plans for optimising this generator.